src/WebJobs.Script/Config/ConsoleLoggingOptionsSetup.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    public class ConsoleLoggingOptionsSetup : IConfigureOptions<ConsoleLoggingOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public ConsoleLoggingOptionsSetup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(ConsoleLoggingOptions options)
+        {
+            options.LoggingDisabled = _configuration.GetValue<int>(EnvironmentSettingNames.ConsoleLoggingDisabled) == 1;
+
+            int bufferSize = _configuration.GetValue<int>(EnvironmentSettingNames.ConsoleLoggingBufferSize, ConsoleLoggingOptions.DefaultBufferSize);
```

Comment:
While this is fine, it does make the value of `BufferSize` a bit subjective to the order of multiple `.Configure` calls. Some recommendations on improving it:

1. Instead of setting to default here, just set to default in the options object itself (`public int BufferSize { get; set; } = 8000;`), then would need a TryGet style approach for reading the value - setting only if it is present (and not mistakenly setting 0 when it is not present).
2. OR - make `BufferSize` an `int?` and set it to default only if it is still `null`.

Reply:
OK, added a test case that demonstrates the problem and made it pass by checking if the setting is present.